tools/mtouch/mtouch.cs

```diff
@@ -878,6 +878,10 @@ public static bool CanWeSymlinkTheApplication (Application app)
 			if (app.EnvironmentVariables.Count > 0)
 				return false;
 
+			// If we are asked to run with concurrent sgen we also need to pass environment variables
```

Comment:
It's correct insofar as saying that concurrent gc is enabled using an environment variable.

This particular variable (MONO_GC_PARAMS) is not in app.EnvironmentVariables (which only contains environment variables passed to mtouch using --setenv), which is why the previous condition does not trigger.